CRM/Report/Form.php

```diff
@@ -4957,4 +4959,22 @@ public function getPermissionedFTClauseForLineItem(&$clauses, $alias = NULL) {
     $clauses[] = $alias . '.' . $clause;
   }
 
+  /**
+   * Function to build select where clause for financial type ACL.
+   *
+   * @param object $query
+   *
+   */
+  public function getSelectWhereClause(&$query) {
+    $clauses = array();
+    $aliases = $query->getVar('_aliases');
+    CRM_Utils_Hook::selectWhereClause('aclReport', $clauses);
+    if (!empty($clauses)) {
+      foreach ($clauses as $field => $vals) {
+        $tableAlias = $aliases[$field];
```

Comment:
I'm looking at the other functions
CRM_Core_DAO::getSelectWhereClause
and CRM_Utils_SQL::mergeSubquery
and wondering if this could be merged with them or leverage them.

My thinking at the moment is that it would be worth adding code comments that those exist and that people developing this area further should re-review that, but putting the effort now into adding unit tests to lock in this working.